packages/jest-snapshot/src/index.js

```diff
@@ -10,96 +10,92 @@
 'use strict';
 
 import type {HasteFS} from 'types/HasteMap';
-import type {Jasmine} from 'types/Jasmine';
 import type {Path} from 'types/Config';
-import type {SnapshotState} from './SnapshotState';
-
-const SnapshotFile = require('./SnapshotFile');
 
+const SnapshotState = require('./State');
+const {SnapshotFile, SNAPSHOT_EXTENSION} = require('./SnapshotFile');
+const diff = require('jest-diff');
 const fileExists = require('jest-file-exists');
 const fs = require('fs');
-const matcher = require('./matcher');
 const path = require('path');
 
-const EXTENSION = SnapshotFile.SNAPSHOT_EXTENSION;
+const EXTENSION = SNAPSHOT_EXTENSION;
 
-const patchAttr = (attr, state) => {
-  attr.onStart = function(onStart) {
-    return function(context) {
-      state.setSpecName(context.getFullName());
-      state.setCounter(0);
-      if (onStart) {
-        onStart(context);
+const cleanup = (hasteFS: HasteFS, update: boolean) => {
+  const pattern = '\\.' + EXTENSION + '$';
+  const files = hasteFS.matchFiles(pattern);
+  const filesRemoved = files
+    .filter(snapshotFile => !fileExists(
+      path.resolve(
+        path.dirname(snapshotFile),
+        '..',
+        path.basename(snapshotFile, '.' + EXTENSION),
+      ),
+      hasteFS,
+    ))
+    .map(snapshotFile => {
+      if (update) {
+        fs.unlinkSync(snapshotFile);
       }
-    };
-  }(attr.onStart);
+    })
+    .length;
+
+  return {
+    filesRemoved,
+  };
 };
 
-const patchJasmine = (jasmine, state) => {
-  jasmine.Spec = (realSpec => {
-    const Spec = function Spec(attr) {
-      patchAttr(attr, state);
-      realSpec.call(this, attr);
-    };
-    Spec.prototype = realSpec.prototype;
-    for (const statics in realSpec) {
-      if (Object.prototype.hasOwnProperty.call(realSpec, statics)) {
-        Spec[statics] = realSpec[statics];
-      }
-    }
-    return Spec;
-  })(jasmine.Spec);
+let snapshotState;
+
+const initSnapshotState = (testFile: Path, update: boolean) => {
+  snapshotState = new SnapshotState(testFile, update);
+  return snapshotState;
+};
+
+const getSnapshotState = () => snapshotState;
+
+const matcher = function(received: any) {
+  this.dontThrow();
+  const {currentTestName, isNot, snapshotState} = this;
+
+  if (isNot) {
+    throw new Error(
+      'Error: Jest: `.not` can not be used with `.toMatchSnapshot()`.',
+    );
+  }
+
+  if (!snapshotState) {
+    throw new Error('snapshot state should be initialized.');
```

Comment:
This is a simple invariant, right? It should only happen if someone tries to use this package outside of Jest and hasn't initialized everything properly, right?

`Jest: snapshot state must be initialized.`

You have a habit of using `should` for things that are `musts`, just figured I should point it out because I've been correcting it a bunch. Together you and I might actually become good at English one day if we keep helping each other.